### ISSUE TITLE
Cannot import module in subfolder when connected through remote

### ISSUE THREAD

**Issue by guidocioni** [microsoft/vscode-jupyter#15777]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

When trying to import a file located into a subdirectory of the folder where the notebook is running I get this error

<img width="634" alt="Screenshot 2024-06-14 at 13 50 26" src="https://github.com/microsoft/vscode-jupyter/assets/12760310/fff4616d-0269-492f-a060-44a6b09e9b63">

I know the import is supposed to be working because:

1. when running the same code into a Python file everything is working as expected
2. it used to work until a few weeks ago, so probably something has been introduced in the new version of either the Jupyter extension or Vscode.

I've been trying to narrow down the cause of this and I'm at a lost. It seems the `PATH` environmental variable is correctly sourced and the correct Python version is used. I tried to select multiple times the Python interpreter, restart kernel, restart server...

Curiously enough, if I connect to the Jupyter server that VsCode is itself starting (as shown in the Output:Jupyter tab), I'm indeed able to run the code without issues (VsCode has automatically forwarded the port 8888 locally from the remote host)

<img width="903" alt="Screenshot 2024-06-17 at 19 03 23" src="https://github.com/microsoft/vscode-jupyter/assets/12760310/9976df88-8a65-4b02-8334-ec4ddcc7b648">

This tells me that the Jupyter server spawned by VsCode is indeed correct, but the kernel that VsCode is spawning have something wrong. I tried to investigate a simple `!printenv` on both Jupyter running on `localhost:8888` and Jupyter running in Vscode, and I've indeed seen some differences, but nothing stands out as problematic. Furthermore, I wouldn't really know how to tinker with the way VsCode starts the kernel (because I don't know where this is happening).

I can only reproduce this issue with a remote Machine running CentOS, I didn't have it on more recent machines running Ubuntu. 

### VS Code Version

1.90.1

### Jupyter Extension Version

v2024.5.0

### Jupyter logs

```shell
Visual Studio Code (1.90.1, ssh-remote, desktop)
Jupyter Extension Version: 2024.5.0.
Python Extension Version: 2024.8.1.
Pylance Extension Version: 2024.6.1.
Platform: linux (x64).
Workspace folder /var/models/wnetwork_new, Home = /home/cioni
19:07:08.576 [warn] Exception while attempting zmq : /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/node_modules/zeromq/prebuilds/linux-x64/node.napi.glibc.node)
19:07:08.577 [warn] Exception while attempting zmq (fallback) : /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/node_modules/zeromqold/prebuilds/linux-x64/node.napi.glibc.node)
19:07:09.185 [info] Attempting to start a server because of preload conditions ...
19:07:09.191 [debug] Start refreshing Kernel Picker (1718644029191)
19:07:11.451 [warn] No interpreter with path ~/miniconda3/envs/stations_new/bin/python found in Python API, will convert Uri path to string as Id ~/miniconda3/envs/stations_new/bin/python
19:07:11.454 [debug] Get Custom Env Variables, Class name = Bm, completed in 2264ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunPythonCode"
19:07:11.454 [debug] Jupyter Paths /kernels: 
19:07:11.455 [debug] Kernel Spec Root Paths, /usr/share/jupyter/kernels, /usr/local/share/jupyter/kernels, ~/.local/share/jupyter/kernels
19:07:11.770 [debug] End refreshing Kernel Picker (1718644029191)
19:07:13.532 [debug] Start refreshing Kernel Picker (1718644033532)
19:07:13.639 [debug] Get Custom Env Variables, Class name = Bm, completed in 6ms, has a truthy return value, Arg 1: "/var/models/wnetwork_new", Arg 2: "RunPythonCode"
19:07:13.898 [debug] End refreshing Kernel Picker (1718644033532)
19:07:13.928 [debug] KernelProvider switched kernel to id = .jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher
19:07:13.929 [debug] start the kernel, options.disableUI=true for /var/models/wnetwork_new/scripts/test.ipynb
19:07:13.994 [debug] Controller selection change completed
19:07:14.000 [info] Starting Kernel (Python Path: ~/miniconda3/envs/stations_new/bin/python, Conda, 3.11.7) for '/var/models/wnetwork_new/scripts/test.ipynb' (disableUI=true)
19:07:15.291 [debug] Got env vars from Python Ext in 1658ms for ~/miniconda3/envs/stations_new/bin/python, with env var count 83.
19:07:15.293 [debug] Getting activated env variables, Class name = Pp, completed in 1660ms, has a truthy return value, Arg 1: "/var/models/wnetwork_new", Arg 2: "~/miniconda3/envs/stations_new/bin/python", Arg 3: undefined
19:07:15.309 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python -c "import jupyter;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
19:07:15.321 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python -c "import notebook;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
19:07:15.332 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python -m pip list
19:07:15.370 [debug] Checking if product is installed, Class name = ND, completed in 1817ms, has a truthy return value, Arg 1: 18, Arg 2: undefined, Return Value: true
19:07:15.378 [debug] Checking if product is installed, Class name = ND, completed in 1823ms, has a truthy return value, Arg 1: 20, Arg 2: undefined, Return Value: true
19:07:15.388 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python -m jupyter kernelspec --version
19:07:15.689 [debug] Interpreter for Pylance for Notebook URI "/var/models/wnetwork_new/scripts/test.ipynb" is ~/miniconda3/envs/stations_new/bin/python
19:07:16.010 [debug] Starting notebook server.
19:07:16.011 [debug] Launching server
19:07:16.014 [info] Starting Jupyter Server
19:07:16.017 [debug] Get Custom Env Variables, Class name = Bm, completed in 1ms, has a truthy return value, Arg 1: "/var/models/wnetwork_new", Arg 2: "RunPythonCode"
19:07:16.018 [debug] Got env vars from Python Ext in 2ms for ~/miniconda3/envs/stations_new/bin/python, with env var count 83.
19:07:16.019 [debug] Getting activated env variables, Class name = Pp, completed in 3ms, has a truthy return value, Arg 1: undefined, Arg 2: "~/miniconda3/envs/stations_new/bin/python", Arg 3: undefined
19:07:16.026 [debug] Generating custom default config at /tmp/09c5957b-dfbc-419b-9c46-0c9915c96e7f/jupyter_notebook_config.py
19:07:16.027 [debug] Starting Jupyter from ~/miniconda3/envs/stations_new/bin/python with arguments --no-browser --notebook-dir="/var/models/wnetwork_new" --config=/tmp/09c5957b-dfbc-419b-9c46-0c9915c96e7f/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0
19:07:16.046 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python -m jupyter notebook --no-browser --notebook-dir="/var/models/wnetwork_new" --config=/tmp/09c5957b-dfbc-419b-9c46-0c9915c96e7f/jupyter_notebook_config.py --NotebookApp.iopub_data_rate_limit=10000000000.0
19:07:16.047 [debug] Waiting for Jupyter Notebook
19:07:18.943 [debug] [I 2024-06-17 17:07:18.941 ServerApp] jupyter_lsp | extension was successfully linked.
19:07:18.952 [debug] [I 2024-06-17 17:07:18.951 ServerApp] jupyter_server_terminals | extension was successfully linked.
19:07:18.956 [debug] [W 2024-06-17 17:07:18.955 LabApp] 'iopub_data_rate_limit' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
19:07:18.968 [debug] [W 2024-06-17 17:07:18.967 ServerApp] ServerApp.iopub_data_rate_limit config is deprecated in 2.0. Use ZMQChannelsWebsocketConnection.iopub_data_rate_limit.
19:07:18.969 [debug] [I 2024-06-17 17:07:18.967 ServerApp] jupyterlab | extension was successfully linked.
19:07:18.979 [debug] [I 2024-06-17 17:07:18.978 ServerApp] notebook | extension was successfully linked.
19:07:19.583 [debug] [I 2024-06-17 17:07:19.582 ServerApp] notebook_shim | extension was successfully linked.
19:07:19.610 [debug] [I 2024-06-17 17:07:19.609 ServerApp] notebook_shim | extension was successfully loaded.
19:07:19.620 [debug] [I 2024-06-17 17:07:19.618 ServerApp] jupyter_lsp | extension was successfully loaded.
19:07:19.629 [debug] [I 2024-06-17 17:07:19.628 ServerApp] jupyter_server_terminals | extension was successfully loaded.
19:07:19.642 [debug] [I 2024-06-17 17:07:19.641 LabApp] JupyterLab extension loaded from /home/~/miniconda3/envs/stations_new/lib/python3.11/site-packages/jupyterlab
19:07:19.647 [debug] [I 2024-06-17 17:07:19.646 LabApp] JupyterLab application directory is /home/~/miniconda3/envs/stations_new/share/jupyter/lab
19:07:19.648 [debug] [I 2024-06-17 17:07:19.647 LabApp] Extension Manager is 'pypi'.
19:07:19.722 [debug] [I 2024-06-17 17:07:19.721 ServerApp] jupyterlab | extension was successfully loaded.
19:07:19.734 [debug] [I 2024-06-17 17:07:19.732 ServerApp] notebook | extension was successfully loaded.
19:07:19.739 [debug] [I 2024-06-17 17:07:19.738 ServerApp] Serving notebooks from local directory: /var/models/wnetwork_new
19:07:19.741 [debug] [I 2024-06-17 17:07:19.741 ServerApp] Jupyter Server 2.13.0 is running at:
19:07:19.743 [debug] [I 2024-06-17 17:07:19.742 ServerApp] http://localhost:8888/tree?token=2c0cfb94c6732ca87677ffcda15f1c6857325d69e42a9461
19:07:19.790 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/pythonFiles/vscode_datascience_helpers/getServerInfo.py
19:07:19.798 [debug] [I 2024-06-17 17:07:19.758 ServerApp]     http://127.0.0.1:8888/tree?token=2c0cfb94c6732ca87677ffcda15f1c6857325d69e42a9461
    [I 2024-06-17 17:07:19.758 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
    [C 2024-06-17 17:07:19.767 ServerApp]
    To access the server, open this file in a browser:
    file:///home/~/.local/share/jupyter/runtime/jpserver-22643-open.html
    Or copy and paste one of these URLs:
    http://localhost:8888/tree?token=2c0cfb94c6732ca87677ffcda15f1c6857325d69e42a9461
    http://127.0.0.1:8888/tree?token=2c0cfb94c6732ca87677ffcda15f1c6857325d69e42a9461
19:07:19.855 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/pythonFiles/vscode_datascience_helpers/getServerInfo.py
19:07:19.947 [debug] [I 2024-06-17 17:07:19.896 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
19:07:20.001 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/pythonFiles/vscode_datascience_helpers/getServerInfo.py
19:07:20.425 [debug] Handle Execution of Cells 58 for /var/models/wnetwork_new/scripts/test.ipynb
19:07:21.389 [debug] Check if a kernel .jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher is usable
19:07:21.391 [info] Check & install missing Kernel dependencies for ~/miniconda3/envs/stations_new/bin/python, ui.disabled=false for resource '/var/models/wnetwork_new/scripts/test.ipynb'
19:07:21.423 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python -c "import ipykernel;print('6af208d0-cb9c-427f-b937-ff563e17efdf')"
19:07:22.131 [debug] Checking if product is installed, Class name = ND, completed in 740ms, has a truthy return value, Arg 1: 19, Arg 2: undefined, Return Value: true
19:07:22.132 [debug] Are Dependencies Installed, Class name = dd, completed in 741ms, has a truthy return value, Arg 1: "~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher", Arg 2: {"a":false,"b":{"w":1,"u":{"i":-1,"end":0},"t":[{"id":1603}]}}, Arg 3: true, Return Value: true
19:07:22.133 [debug] Install Missing Dependencies, Class name = dd, completed in 743ms, has a falsy return value, Arg 1: {"resource":{"$mid":1,"fsPath":"/var/models/wnetwork_new/scripts/test.ipynb","external":"file:///var/models/wnetwork_new/scripts/test.ipynb","path":"/var/models/wnetwork_new/scripts/test.ipynb","scheme":"file"},"kernelConnection":{"id":".jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher","kernelSpec":{"specFile":"/home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/temp/jupyter/kernels/python3117jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c/kernel.json","interpreterPath":"/home/~/miniconda3/envs/stations_new/bin/python","isRegisteredByVSC":"registeredByNewVersionOfExt","name":"python3117jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c","argv":["python","-m","ipykernel_launcher","-f","{connection_file}"],"language":"python","executable":"python","display_name":"stations_new","metadata":{"interpreter":{"path":"/home/~/miniconda3/envs/stations_new/bin/python"}},"env":{}},"interpreter":{"id":"/home/~/miniconda3/envs/stations_new/bin/python","path":"/home/~/miniconda3/envs/stations_new/bin/python"},"kind":"startUsingPythonInterpreter"},"ui":{"_event":{"w":4,"u":{"i":4,"end":4},"t":[{"id":1302},{"id":1316},{"id":1330},{"id":1331}]},"_disableUI":false},"token":{"a":false,"b":{"w":1,"u":{"i":-1,"end":0},"t":[{"id":1603}]}},"ignoreCache":true,"cannotChangeKernels":false}, Return Value: 0
19:07:22.145 [debug] Updating Kernel Environment for .jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher for interpreter /home/~/miniconda3/envs/stations_new/bin/python
19:07:22.161 [debug] Python KernelSpec argv[0] updated for .jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher from '/home/~/miniconda3/envs/stations_new/bin/python' to '~/miniconda3/envs/stations_new/bin/python'
19:07:22.164 [debug] Get Custom Env Variables, Class name = Bm, completed in 1ms, has a truthy return value, Arg 1: "/var/models/wnetwork_new/scripts/test.ipynb", Arg 2: "RunPythonCode"
19:07:22.166 [debug] Got env vars from Python Ext in 2ms for ~/miniconda3/envs/stations_new/bin/python, with env var count 83.
19:07:22.166 [debug] Getting activated env variables, Class name = Pp, completed in 3ms, has a truthy return value, Arg 1: "/var/models/wnetwork_new/scripts/test.ipynb", Arg 2: "~/miniconda3/envs/stations_new/bin/python", Arg 3: undefined
19:07:22.286 [debug] Updated kernel spec for .jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher with environment variables for ~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/temp/jupyter/kernels/python3117jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c/kernel.json
19:07:22.470 [debug] [I 2024-06-17 17:07:22.469 ServerApp] Kernel started: 7e075c4c-98c4-4787-a803-29a8ca4440b6
19:07:22.499 [info] http://localhost:8888: Kernel started: 7e075c4c-98c4-4787-a803-29a8ca4440b6
19:07:24.122 [debug] [I 2024-06-17 17:07:24.110 ServerApp] Connecting to kernel 7e075c4c-98c4-4787-a803-29a8ca4440b6.
19:07:24.275 [info] Started session for kernel startUsingPythonInterpreter:.jvsc74a57bd0fdb8be4ba885e374374c2797cae172193a724b22d3e0bf323b23803166b5f94c./home/~/miniconda3/envs/stations_new/python./home/~/miniconda3/envs/stations_new/python.-m#ipykernel_launcher
19:07:24.424 [debug] Requesting Kernel info
19:07:24.432 [info] Kernel successfully started
19:07:24.436 [debug] getDataDirsImpl, Class name = jr (started execution), Arg 1: "/var/models/wnetwork_new/scripts/test.ipynb", Arg 2: "~/miniconda3/envs/stations_new/bin/python"
19:07:24.437 [debug] Get Custom Env Variables, Class name = Bm, completed in 1ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunPythonCode"
19:07:24.438 [debug] Jupyter Paths : 
19:07:24.453 [info] Process Execution: ~/miniconda3/envs/stations_new/bin/python /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/pythonFiles/printJupyterDataDir.py
19:07:24.489 [debug] Kernel acknowledged execution of cell 58 @ 1718644044487
19:07:24.492 [debug] getDataDirsImpl, Class name = jr, completed in 55ms, has a truthy return value, Arg 1: "/var/models/wnetwork_new/scripts/test.ipynb", Arg 2: "~/miniconda3/envs/stations_new/bin/python", Return Value: <Uri:/home/~/.local/share/jupyter>, <Uri:/home/~/miniconda3/envs/stations_new/share/jupyter>, <Uri:/usr/local/share/jupyter>, <Uri:/usr/share/jupyter>
19:07:24.873 [warn] Cell completed with errors [hd [Error]: No module named 'lib'
	at r.execute (/home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/extension.node.js:307:4958)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)] {
  ename: 'ModuleNotFoundError',
  evalue: "No module named 'lib'",
  traceback: [
    '\x1B[0;31m---------------------------------------------------------------------------\x1B[0m',
    '\x1B[0;31mModuleNotFoundError\x1B[0m                       Traceback (most recent call last)',
    'Cell \x1B[0;32mIn[1], line 1\x1B[0m\n' +
      '\x1B[0;32m----> 1\x1B[0m \x1B[38;5;28;01mfrom\x1B[39;00m \x1B[38;5;21;01mlib\x1B[39;00m\x1B[38;5;21;01m.\x1B[39;00m\x1B[38;5;21;01mdata_utils\x1B[39;00m \x1B[38;5;28;01mimport\x1B[39;00m get_daily\n',
    "\x1B[0;31mModuleNotFoundError\x1B[0m: No module named 'lib'"
  ]
}
```

### Coding Language and Runtime Version

Python 3.11

### Running Jupyter locally or remotely?

Remote

**Comment by krzim** [comment]
I'm also seeing this issue on Windows 11 running VSCode connected to a dev container. I had to roll back to v2024.3.1 as both v2024.4.0 and v2024.5.0 would not let me import local files.

**Comment by guidocioni** [comment]
> I'm also seeing this issue on Windows 11 running VSCode connected to a dev container. I had to roll back to v2024.3.1 as both v2024.4.0 and v2024.5.0 would not let me import local files.

Did you manage to understand what is causing it? I have other projects on the same remote machine where the local import does not fail. Plus, I don't have the issue on other machines...
I really cannot narrow it down, but I have the feeling vscode is unsetting something when starting the kernel

**Comment by krzim** [comment]
I'll be honest, I didn't do much more digging once I unblocked myself. I'm happy to debug if needed though.

**Comment by guidocioni** [comment]
Cannot believe ChatGPT pointed me in the right direction...by making it analyzing the two different files I indeed realized that the `--notebook-dir` argument passed when starting `jupyter` was different.

- Working Log: `--notebook-dir="/var/models/wnetwork_new/scripts"`.
- Issue Log: `--notebook-dir="/var/models/wnetwork_new"`.

It was then clear to understand how this could lead to issue as, in my case, the module I'm trying to import is located in `/var/models/wnetwork_new/scripts/lib`. 

I added this line to the `settings.json` of my workspace only and now everything is working.
```json
    "jupyter.jupyterCommandLineArguments": [
        "--notebook-dir='/var/models/wnetwork_new/scripts'"
    ],
```
Of course this is only a workaround. 
The problem seems so stem from the fact that the notebook file root is not set to `${fileDirname}` so the issue may be related to https://github.com/microsoft/vscode-jupyter/issues/8771

**Comment by DonJayamanne** [comment]
Thank you for filing this issue and sorry got not getting back sooner.

> 19:07:08.576 [warn] Exception while attempting zmq : /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/node_modules/zeromq/prebuilds/linux-x64/node.napi.glibc.node)
19:07:08.577 [warn] Exception while attempting zmq (fallback) : /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/node_modules/zeromqold/prebuilds/linux-x64/node.napi.glibc.node)
19:07:09.185


I understand there's an issue with the file directory. however I believe the above is part of the problem
Is it possible for you to somehow install GBLIC on your machine.

**Comment by guidocioni** [comment]
> Thank you for filing this issue and sorry got not getting back sooner.
> 
> > 19:07:08.576 [warn] Exception while attempting zmq : /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/node_modules/zeromq/prebuilds/linux-x64/node.napi.glibc.node) 19:07:08.577 [warn] Exception while attempting zmq (fallback) : /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /home/~/.vscode-server/extensions/ms-toolsai.jupyter-2024.5.0-linux-x64/dist/node_modules/zeromqold/prebuilds/linux-x64/node.napi.glibc.node)
> > 19:07:09.185
> 
> I understand there's an issue with the file directory. however I believe the above is part of the problem Is it possible for you to somehow install GBLIC on your machine.

Hi,
my previous comment shows that glibc is not part of the problem. This is related to the issues I linked.
I cannot install glibc because this is a.system running on an old centos version.

**Comment by DonJayamanne** [comment]
The reason I suggested installing glib is because there is a big difference in the way the kennels are started and the cwd is setup
